python/tank/util/version.py

```diff
@@ -7,34 +7,51 @@
 # By accessing, using, copying or modifying this work you indicate your
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
-import warnings
 import contextlib
 import sys
+import warnings
 
-LooseVersion = None
+version_parse = None
 try:
     import packaging.version
+
+    version_parse = packaging.version.parse
 except ModuleNotFoundError:
     try:
         # Try importing from setuptools.
         # If it fails, then we can't do much at the moment
         # The DCC should have either setuptools or packaging installed.
         from setuptools._distutils.version import LooseVersion
+
+        version_parse = LooseVersion
     except ModuleNotFoundError:
         try:
-            # DCCs with older versions of Python 3.12
             from distutils.version import LooseVersion
+
+            version_parse = LooseVersion
         except ModuleNotFoundError:
-            pass
+            version_parse = str
 
-from . import sgre as re
 from .. import LogManager
 from ..errors import TankError
-
+from . import sgre as re
 
 logger = LogManager.get_logger(__name__)
 GITHUB_HASH_RE = re.compile("^[0-9a-fA-F]{7,40}$")
 
+# Normalize non-standard version formats (e.g., "v1.2.3", "6.3v6", "1.2-3")
+# into PEP 440–compliant forms ("1.2.3") to ensure compatibility with
+# Python’s version parsing utilities (e.g., packaging.version.parse).
+# Reference: https://peps.python.org/pep-0440/
+_VERSION_PATTERNS = [
+    (re.compile(r"^v(\d+)\.(\d+)\.(\d+)$"), r"\1.\2.\3"),  # v prefix: v1.2.3 -> 1.2.3
```

Comment:
In my experience, we don't need to change this syntax, packaging.version is OK with it.